### ISSUE TITLE
How to change the flight behavior 

### ISSUE THREAD

**Issue by notsolowki** [mefistotelis/phantom-firmware-tools#15]
i figured it out

**Comment by notsolowki** [comment]
i should have said there are 2 possibilities. 1 being the above message, 2 being that install i installed the fw with p3x debug. i maybe even leaning more towards the p3x debug. let me know!! now you cann all edit and crash your drones.

**Comment by notsolowki** [comment]
@mefistotelis

**Comment by notsolowki** [comment]
@ferraript you should make a GUI to edit flyc_param. once all the parameters are defined. would be cool. i imagine it would be able to laod both flc_params. one from the app and one from the firmware. then make changes to both accordingly

**Comment by notsolowki** [comment]
More notes: i didnt have to change anything in the app. just had to flash with p3x_debug

**Comment by notsolowki** [comment]
back to the app having nothing to do with it

**Comment by ferraript** [comment]
I'm lost in your thoughts once again :smile:
So is it possible to push params from app to AC or not?

>you should make a GUI to edit flyc_param

unfortunately, I'm neither Android nor Java developer

**Comment by notsolowki** [comment]
you dont have to push them from the app

**Comment by notsolowki** [comment]
you just have to install the fw with the p3x debug. example. WITH 1.7.6 INSTALLED modify the 1.7.6 flight controller binary to your liking. then use the p3x_debug to OVERWRITE 1.7.6 with 1.7.6

**Comment by notsolowki** [comment]
i might add that it take 23 minutes to flash the fw with this method. it dont give very many indicators but it will make the update complete sound when its done. maybe even you could edit airport mode this was idk

**Comment by notsolowki** [comment]
im still testing the parameters. but i can tell you that basic.craft.thrust makes it go from idle to max motor speed right away. if you set it too high it dances in the air lol

**Comment by notsolowki** [comment]
and @ferraript, didnt you wirte that program to convert the logs? if you could make a windows gui like that. one that can read the parameters in the flyc_param file and list them in the gui. they would have boxes for the max,min,default value. each would have its own description. then it could save the changes to the json file.. right now im trying to figure out how the parameters behave so we can get some descriptions. unless someone else has the descriptions.. that would be cool

**Comment by notsolowki** [comment]
okay another update: the parameters do have to be restored in the app. so the apps flyc and the firmwares flc have to be updated

**Comment by notsolowki** [comment]
i can confirm my takeoff speed is 27 mph which is 12 meters per second

**Comment by mefistotelis** [comment]
> i might add that it take 23 minutes to flash the fw with this method

If you're updating the whole firmware - note that you could update only the flyc module, would be faster and safer.

**Comment by notsolowki** [comment]
i was wondering that. could i just delete all the un-needed modules when repacking. and dont it upgrade the mi01 binary the normal way? i wonder why it only works when i use the debug file. maybe ill try it the normal way again just to confirm.

**Comment by mefistotelis** [comment]
No, that's not how it works.

You do not need the packing operation at all, you just need to rename the module you want to update. Correct file names are listed in the downgrade discussuin #4 .

**Comment by notsolowki** [comment]
so
modules=0 mi01 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0  ??

**Comment by mefistotelis** [comment]
No. File name.

**Comment by notsolowki** [comment]
so i should rename mi01 to m0305.bin?

**Comment by mefistotelis** [comment]
No. [correct file name for m0306](https://github.com/mefistotelis/phantom-firmware-tools/issues/4#issuecomment-272307530)

Why do you mention m03s05? it is encrypted, so see my comment above for encrypted modules.

**Comment by notsolowki** [comment]
i figured since if was the first one in the list it would be the first module. i forgot theres mi00. so i can take mi01.bin and rename it m0305.bin and place it on the sdcard? would i use the debug file too

**Comment by notsolowki** [comment]
would be great to save some time because at this rate ill be testing parameters for a month lol

**Comment by mefistotelis** [comment]
Please read the post which I linked to.

**Comment by notsolowki** [comment]
i know im sorry i just did that. so i can rename mi01 to PMCAPPFw3.bin?

**Comment by notsolowki** [comment]
so i cant take mi01 and rename it to PMCLDRFw3.bin and place it on the sdcard?

**Comment by mefistotelis** [comment]
Yes. Placing this file on SD card should initiate the update. I don't think the *_debug file is needed, but you have to check that.

**Comment by notsolowki** [comment]
okay we got PMCLDRFw3 and PMCLDRAPP, Whats the APP for?

**Comment by notsolowki** [comment]
and another question. how come it would update the height but nothing else before when doing the upgrade the normal way. but now if i use the debug file they all work? i might be working ahead of myself here

**Comment by mefistotelis** [comment]
Do not update with both. read everything again.

**Comment by notsolowki** [comment]
i know i was just asking whats the difference between the two. i didnt know what the ldr app was for

**Comment by mefistotelis** [comment]
> can i rename mi01 to PMCLDRFw3.bin

No. This is not correct and will brick your drone.

> and it will only write mi01.bin

If you will not provide any other file on the card, then no other modules will be updated

>  will it have and update finished indicator

I don't know.

**Comment by notsolowki** [comment]
then i dont get what your saying., can you just show me what you mean

**Comment by notsolowki** [comment]
so instead if  PMCLDRFw3.bin its m0306.bin?

**Comment by notsolowki** [comment]
i asked you this in an above post and you said it was the correct way. thats why im confused

**Comment by mefistotelis** [comment]
> i asked you this in an above post and you said it was the correct way

Your post looked different when I commented ;)

You seem to be confusing modules. Be sure everything is right before you try to do the update.

**Comment by notsolowki** [comment]
okay so  PMCLDRFw3.bin is what i need to rename my modified mi0.bin1 to? thank you

**Comment by mefistotelis** [comment]
No. You're still confusing modules.

**Comment by notsolowki** [comment]
wow im sorry lol, i mean PMCAPPFw3.bin is this correct. change mi01 to PMCAPPFw3.bin

**Comment by mefistotelis** [comment]
Yes, P3X_FW_V01.07.0060_mi01.bin -> MCAPPFw3.bin.

**Comment by notsolowki** [comment]
yes without t P thank you for reminding me that. thank you!!!!!

**Comment by mefistotelis** [comment]
No, sorry, that's my mistake. With P.

**Comment by notsolowki** [comment]
lol thankyou

**Comment by ferraript** [comment]
why don't you just create yourself firmware pack containing m0306 module only and flash it?

**Comment by notsolowki** [comment]
yea thats what we were discussing

**Comment by notsolowki** [comment]
i wonder how fast this drone can descend without vrs

**Comment by ferraript** [comment]
no, you are renaming P3X_FW_V01.07.0060_mi01.bin to MCAPPFw3.bin
and I'm suggesting:
1. extract all modules from firmware using dji_fwcon.py (if you haven't done it already)
2. delete all files except head.ini, mi01.ini and mi01.bin
3. update flight parameters using dji_flyc_param_ed.py
4. rename mi01.* to mi00.*
5. create firmware file using dji_fwcon.py and flash it

**Comment by ferraript** [comment]
>didnt you wirte that program to convert the logs?

I did, but it's written in Delphi / Pascal

>i wonder how fast this drone can descend without vrs

exactly, changing descending speed may be dangerous
I was also thinking of changing those 3 m/s to something higher, but I just like my P3A too much to lose it

**Comment by ferraript** [comment]
>i can confirm my takeoff speed is 27 mph which is 12 meters per second

1. as vertical ascending speed?
2. you achieved that by changing vert_up_vel_0 or something else?
3. I'd like to see DAT log containing this flight, any chance of uploading it somewhere?

**Comment by notsolowki** [comment]
@mefistotelis PMCAPPFw3.bin  PERFECT :)

**Comment by chazzzzz** [comment]
People following this thread will brick their drone...

**Comment by notsolowki** [comment]
hows that

**Comment by notsolowki** [comment]
@mefistotelis have you tried changing your ascend speed?

**Comment by notsolowki** [comment]
i candescend at 12-13 mph with no issues so far

**Comment by chazzzzz** [comment]
Because it boggled my mind and I somewhat understand what is going on... I just hope they read all the way through before attempting this..

We are modifying target m0306 through mi01.bin and renaming it to PMCAPPFw3.bin...Is this Correct?

**Comment by notsolowki** [comment]
if you read everything im sure you will figure it out

**Comment by notsolowki** [comment]
i wouldnt do this unless u got a p3 pro...

**Comment by notsolowki** [comment]
okay 
1. get dji go 2.4.1
2. modify flyc params in mi01
3.repack mi01 with flyc_param_ed.py
4.flash mi01.bin
5. edit the apk and make same said changes to flyc_param
6. with everything powered on goto setting and restore defualt values....
there are you happy

**Comment by notsolowki** [comment]
because i know i am

**Comment by notsolowki** [comment]
dont change any _basic.craft thrust or control.thrust . they are gain settings setting them higer will make it really unstable

**Comment by notsolowki** [comment]
the esc i think still governs the rpm. so you cant get too crazy. you should hold your drone down and test your settings as much as you can before loosing it in the air

**Comment by chazzzzz** [comment]
How do you edit the apk? or can you direct me to the thread that explains this? Also is there anything there to edit RTH speed...Its really all I need...

**Comment by notsolowki** [comment]
i havnt change the apk. i use apk editor pro its really easy

**Comment by notsolowki** [comment]
you will find a flyc_param_infos   inside the dji go app. its under /res/raw/flyc_param_infos

**Comment by notsolowki** [comment]
be careful i already had to emergency land and crash once

**Comment by notsolowki** [comment]
i there theres some setting related to the current of the battery that limit the up velocity. not the settings for low battery but something else. i get 27 mph "up" on a full battery and like 22 on a half to 3qtr

**Comment by notsolowki** [comment]
okay i descended like 30 times at 18 mph with no issues just give it room to stop lol

**Comment by notsolowki** [comment]
some how earlier i was ascending at 27mph now i get a constant 22.4 mph

**Comment by notsolowki** [comment]
@mefistotelis have you managed to change the pitch or get passed 22.4 mph ascend speeds?????

**Comment by mefistotelis** [comment]
I never tried.

**Comment by notsolowki** [comment]
theres no parameter in the json file that points to go home speed i know in the binary its 7.8 am i just missing it?

**Comment by notsolowki** [comment]
you should try it. it seems safe

**Comment by notsolowki** [comment]
@chazzzzz any luck

**Comment by notsolowki** [comment]
@mefistotelis is there any chance i could get you to write a script to edit the go home speed?!?! pleaseeeee!!!

**Comment by chazzzzz** [comment]
Why don't some of the index numbers in the FW1.07 match the flyc_param_infos file in DJI go?

Example:
**FW1.07**
"index" : 180,
		"typeID" : 5,
		"size" : 2,
		"attribute" : 11,
		"minValue" : 20,
		"maxValue" : 500,
		"defaultValue" : 90,      
		"name" : "g_config.control.basic_craft_pitch_0",
		"modify" : true

**DJI Go 2.9.1 flyc_param_infos file**
"index" : 180,
		"typeID" : 5,
		"size" : 2,
		"attribute" : 11,
		"minValue" : 70,
		"maxValue" : 130,
		"defaultValue" : 100,        
		"name" : "g_config.control.basic_roll_0",
		"modify" : true

Is this normal?

**Comment by notsolowki** [comment]
the app dont have all of the same parameters. if i were you i would suggest using dji go 2.5 unless you have success with 2.9?!?

**Comment by chazzzzz** [comment]
Well I don't think I can have success if the indexes don't match up? i wouldn't know what to change... do they match up with 2.5? can you compare these in your version of dji go and the 1.07 bin? Next question is can you just change these values in the go app and let the bin alone?

**Comment by notsolowki** [comment]
i will compare them for you. to be honest i never checked the index numbers. you have to edit the flight controller if the value you want to use in the app is higher than it is in the flight controller. for example the vel_up param is for ascend speed. set the max to 16. then in the apps flyc set it to 16 max and set the default to 16. then int the app where the gain settings are you will see restore defaults. and thats it. i will ad that in dji go 3.1 when you hit reset it says its going to reset the gains/expo but in 2.4 it clearly says its going to set all the mc parameters..

**Comment by notsolowki** [comment]
if you dont resign the app it will uninstall after you reset your phone. the easiest way to reset the app is to extract flc_param from the app. edit it the way you want. then use apk editor PRO and select choose from app. then select simple edit. then goto res/raw and click flyc_param it will ask you to choose the file to replace it with at this point choose your edited flc_param. then after its done "working" it will give you the option to install it as an update. do that and your done.

**Comment by notsolowki** [comment]
i have my descend speed at 18 mph no problems yet and ive tested the heck out of it

**Comment by notsolowki** [comment]
dont worry about brick, i already did that with PMCAPPFw3.bin and it was easy to recover. easy as restarting the update and having to sit through a really annoying beep

**Comment by notsolowki** [comment]
and chazz my indexes are the same way.
 @mefistotelis is this normal ?

**Comment by chazzzzz** [comment]
Yeah i noticed it wont let you edit it until you extract it somewhere else and i did get as far as understanding how to replace it with the edited one but i didn't change anything because i didn't know what to edit because the index #'s don't match so i stopped...I may just try editing in dji go only at some point but now time for bed....

**Comment by notsolowki** [comment]
i can send you a copy of my fw with ascend and descend edited and send you an update apk for 2.4 if youd like

**Comment by chazzzzz** [comment]
not yet but maybe later... thank you for the offer!

**Comment by MrBurnsAT** [comment]
@notsolowki  i would try it   ;-)

**Comment by mefistotelis** [comment]
> Why don't some of the index numbers in the FW1.07 match the flyc_param_infos file in DJI go?

The dji_flyc_param_ed.py does not use indexes at all. I guess the params in DJI Go are gathered from multiple drone types.

**Comment by notsolowki** [comment]
i thought the list in the firmware was longer?!? and is any of you want to use my fw for a phantom 3 PRO this one has edited ascend speeds,idle speed, descend speed. enjoy and let me know if you like it!

instructions:
1. drop PMCAPPFw3.bin on your drones sdcard and update YOUR CURRENTLY INSTALLED 1.7.6
2. install dji go apk 2.4.2
3. install the gen_signed.apk after you install dji go 2.4.2
4. have fun

http://www.mediafire.com/file/re73qwn3nbnyjgn/p3xfw.zip

**Comment by MrBurnsAT** [comment]
Ive downloaded Ur Firmware now.

Must install 1.7 bevore.
Ill try it in the next days. ;-)

**Comment by notsolowki** [comment]
yes you must have 1.7.6 installed before you install that module

**Comment by notsolowki** [comment]
what version do you have now?

**Comment by notsolowki** [comment]
DO NOT USE THE P3X DEBUG TO INSTALL THE MODULE

**Comment by notsolowki** [comment]
you guys want to see it in action?? https://youtu.be/g7Mzy7E6E-I

**Comment by MrBurnsAT** [comment]
WoW Thats really fast

I know that i must install 1.7.6 bevore and that i dont use debug file

**Comment by notsolowki** [comment]
@mefistotelis should i put a link to here in the video description?

**Comment by mefistotelis** [comment]
Sure, you can do that. Having more visibility might get us more skilled people here.

**Comment by chazzzzz** [comment]
"then in the app where the gain settings are you will see restore defaults"

The only restore default settings I could find is for the gimble...I looked in the gain settings and it isn't there...Am i missing something?

**Comment by notsolowki** [comment]
dji """""""""2.4.2""""""""" you will find restore settings. hit them both if you cant figure it out

**Comment by notsolowki** [comment]
Im going to assume your not using the app i said to use

![screenshot_20170125-132728](https://cloud.githubusercontent.com/assets/24588286/22305728/3c4be9d4-e302-11e6-935c-5b1142e0fde3.png)

**Comment by chazzzzz** [comment]
Your assumption is correct...BUT I assumed that when you said this "i will ad that in dji go 3.1 when you hit reset it says its going to reset the gains/expo " that there would be some sort of reset other then the one for the gimble...No offense but I'm more of a "trust but verify" type of person...I would prefer to use a newer version of DJI go if possible...Also do we really even know if you have to reset anything or maybe it just works by changing the default settings in the flyc_prama_infos file in the go app with no reset required?

**Comment by notsolowki** [comment]
i thumbs down that for ignorance

**Comment by notsolowki** [comment]
i give you step by step instructions and you still cant figure it out

**Comment by chazzzzz** [comment]
You’re a funny man!  Hey, I’ll try follow your instruction next time I want to go up really fast. :)

**Comment by notsolowki** [comment]
You need to sign the apk.

**Comment by notsolowki** [comment]
Try using apk editor to extract the file from tge app you want to change,
then use apk editor to repack and install

On Jul 28, 2017 2:15 PM, "phudd" <notifications@github.com> wrote:

I have followed these instructions and have success until i try to install
gen_signed.apk
for some reason it acts like it is installing and then i get app not
installed. i have dji go 2.4.2 installed. I am using samsung galaxy s7edge
not rooted. have tried everything i can think of any help would be greatly
appreciated.

drop PMCAPPFw3.bin on your drones sdcard and update YOUR CURRENTLY
INSTALLED 1.7.6
install dji go apk 2.4.2
install the gen_signed.apk after you install dji go 2.4.2
open dji go and restore default settings
have fun

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/mefistotelis/phantom-firmware-tools/issues/15#issuecomment-318739575>,
or mute the thread

**Comment by phudd** [comment]
this worked thank you 
was wondering if anyone had a link to PMCAPPFw3.bin with the speed and no height restriction or nfz .  thanks again

**Comment by fineluser** [comment]
I just  cann't downgrade to 1.7.6, I am P3A with 1.9.6.when I downgrade to 1.8.8 the compass wrong,then downgrade to 1.7.6 with the P3S_DEBUG TEXT, The YOMO worked wrong,Can someone help me to downgrade to 1.7.6

**Comment by fineluser** [comment]
P3A怎么降级呢，从1.9.6降到1.8.8指南针就错误，再用DEBUG文件降级到1.7.6云台相机就出故障了 求大神支招阿，另外破解固件刷机以后，手机APP上500米的限制还在，不改的话能飞500米以上高吗
